React Native/Styling.md

```diff
@@ -0,0 +1,215 @@
+
+To create styles, the API implements a **`create`** method.
+
+```javascript
+const styles = StyleSheet.create({
+	container:  { borderRadius:  4, borderWidth:  0.5, borderColor:  '#d6d7da'  },
+	title:  { fontSize:  19, fontWeight:  'bold'  },
+	activeTitle:  { color:  'red'  }
+});
+```
+
+```javascript
+<View style={styles.container}>
+	<Text style={styles.title} />
+</View>
+```
+
+**Note:** Styles should always be defined outside of the `render` function.
+
+```javascript
+import React from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+
+export default App = () => (
+  <View style={container}>
+    <Text style={text}>React Native</Text>
+  </View>
+);
+
+const page = StyleSheet.create({
+  container: {
+    flex: 1,
+    padding: 24,
+    backgroundColor: '#fff',
+  },
+  text: {
+    fontSize: 30,
+    color: '#000'
+  },
+});
+```
+
+More on the `Stylesheet API` can be found on:
+
+[https://reactnative.dev/docs/stylesheet](https://reactnative.dev/docs/stylesheet)
+
+Important thing to recognize here is that all measurement values are written as numbers, not pixels like we are used to in web development. That is because all dimensions in React Native are unitless, and represent density-independent pixels. Setting dimensions this way is common for components that should always render at exactly the same size, regardless of screen dimensions.
+
+### Flexbox
+RN uses Flexbox algorithm for defining the component layout. Flexbox properties on RN are practicly identical to web with one difference and that is  `flexDirection`. Default value for `flexDirection` property in RN is `column`, where on web it is `row`.
+
+More on Flexbox layout can be found on:
+[https://reactnative.dev/docs/flexbox](https://reactnative.dev/docs/flexbox)
+
+## STYLED-COMPONENTS
+
+Styling in RN can also be done with `styled components` which is more familiar approach for React developers.
+
+When starting a new project **`styled-components`** should be your default styling option as it consumes styling rules which are identical to React, and provides theming.
+
+Currently there are 2 main libraries used:
+1. [Styled components](https://styled-components.com/docs/basics#react-native)
+2. [Emotion](https://emotion.sh/docs/@emotion/native)
+
+There are almost no differences between them, but we decided that `styled-components` should be the option when starting a new project. Main reason for that is theming typings which can easily be defined when using `styled-components` while on the other hand `emotion` library doesn't have that clean solution yet.
```

Comment:
I would rather suggest using RN styling in separate files for every component in combination with a single global file to define application themes (fonts, colors, frequently used styles). Styled-components contain 10 dependencies (about 1.15 MB in total) and must be tag evaluated, pre-processed, and then finally compiled into pure JavaScript bundles for the JS bridge thread to process it. Also, styled-components don't contain platform-specific styles and I'm also not sure if this type of styling in RN will be compatible with some 3rd party components (if used) because of the difference between web and mobile flex property. JUST A SUGGESTION :)

Reply:
I was going through couple of permformance measure blog posts and overview of the same concern you have and basicly everyone had the same conclusion: if the elements are used smart way (for example using ListView instead of ScrollView for rendering large amount of items), styles don't have much impact on performance. Of course, styled-components do have some performance impact but it's not visible, atleast not in the applications we are writing now.
We may discuss if using this is ok in enterprise application with thousand or tens of thousand of items being rendered, but for now this will just bring React / web developers closer to RN.
But all in all I would like that we "define" some rules or maybe our own styleguide for native Stylesheet usage.
@nives113 I will create a sepearate task for this and assign it to me, so when we feel it's needed we can start working on it.

Reply:
Ok, this is fine then 😄